bash_completion

```diff
@@ -2496,9 +2496,24 @@ complete -F _minimal ''
 
 __load_completion()
 {
-    local cmd="${1##*/}" dir compfile
+    local cmd=$1 cmdname=${1##*/} dir compfile
     local -a paths
-    [[ $cmd ]] || return 1
+    [[ $cmdname ]] || return 1
+
+    local backslash=
+    if [[ $cmd == \\* ]]; then
+        cmd=${cmd:1}
+        # If we already have a completion for the "real" command, use it
+        $(complete -p "$cmd" 2>/dev/null || echo false) "\\$cmd" && return 0
+        backslash=\\
+    fi
+
+    # Try to resolve real path to $cmd, and make it absolute
+    local ret realcmd="" origcmd=$cmd
+    if _comp_realcommand "$cmd"; then
+        realcmd=$ret
+        cmd=${realcmd%/*}/$cmdname # basename could be an alias
```

Comment:
I think I fail to understand this line. I naively thought this should be `cmd=$realcmd` because the command name can be different in the real location, such as `~/bin/b -> /usr/bin/bash`. There seems to be a reason related to an alias as in the code comment, but I'm not sure what it exactly is now.

Reply:
"Alias" is actually not a good word to describe this, but I think you got the intent right: the basename of the resolved command can be different from the one we were invoked with.

Installing our completion for the "real" resolved name would be problematic for example in cases where the target command has different behavior depending on the basename it is invoked as. `busybox` is one example of this.

Elaborated on the commentary in 3f479c173b9bd2619962a47b120e48acd54d6c0b

Reply:
Hmm, good point. I suppose we should actually resolve absolute path to the command without doing any symlink resolution, essentially what `_comp_realcommand` now does, but without doing any of the realpath/readlink things?

Reply:
Ah, yes. I think that is the solution.

The original reason that we wanted to resolve the symbolic links is to pick up the corresponding `share/bash-completion/completions`, so the symlink-resolved `realcmd` is still needed. So the conversion of `cmd` to an absolute path should be done independently of `realcmd`.

Reply:
28b587a47058395fdacb7eafe866685af1e74e3c and e4ede514249c4250dd43139f54fa3c46554a1dbc are a quick attempt at fixing this, more or less untested.

Reply:
Need to call it a day now, revisiting later. But do feel free to push changes you see fit here in the meantime if you like.

Reply:
Maybe I have confused you with my previous reply, but I actually agreed to switch to the absolute path without resolving the symbolic links. I wanted to note that the symlink-resolved `realcmd` is still used later for another purpose.